### ISSUE TITLE
[Bug] GetLocationAsync returning old/non-current latitude and longitude 

### ISSUE THREAD

**Issue by Bejasc** [xamarin/Essentials#1315]
### Description

I'm having some trouble with (what appears to be) `GetLocationAsync` returning a cached result, or a result with a _new_ timestamp, but old latitude and longitude values. 

Coming from the [old Geolocator](https://github.com/jamesmontemagno/GeolocatorPlugin) plugin, I recall mention of [only being able to get a location a few times each hour](https://jamesmontemagno.github.io/GeolocatorPlugin/FAQ.html). 

I cannot find any such warning in current documetnation for the Essentials.Geolocation package. 

I am unsure if this is a limitation of the device/package, or something missing in my implementation. 

Regardless, I would like to be able to understand this issue, so I can report back to my Test and BA teams about the existence of any limitations. 

---

#### Code snippet
```
    var request = new GeolocationRequest(GeolocationAccuracy.High);
    var loc = await Geolocation.GetLocationAsync(request);
    //Value of loc is serialized to Json string and processed by server
```

### Steps to Reproduce
Below scenario was experienced on iPhone XS running iOS 13.5.1
No power saving settings were enabled. Wifi was disabled. 

_Position A and Position B refer to different physical areas, in my case, a few blocks down the street._ 

1.
At 12pm, physically at Location A. Call `GetLocationAsync`. Located with my current physical latitude and longitude at Position A. This is current and correct.

2.
At 12:10pm, at Location B. Call `GetLocationAsync`. Located with a _new_ current physical latitude and longitude, at Position B. This is current and correct.

Both results are entirely unique with the Timestamp, Latitude, and Longitude, and considered correct and accurate.

3.
At 12:20pm, now back at Location A. Call `GetLocationAsync`. A brand new result is produced, with a new timestamp. However, Latitude and Longitude are using values from the previous result (Location B), despite the device physically being over 1km away.
_as if it were cached, and "not different enough" to deem actually getting a new loc result?_ 

4.
Any calls placed through GetLocationAsync continue to report the old location, with a new timestamp.

5. (Device fix)
At 12:21pm, open Google Maps or other mapping/locating software on the device. 
The location the user is at _briefly_ is represented as the old location - but soon updates to the current / correct physical device location

6. Calling `GetLocationAsync` after opening Maps - the location is now correctly reported at Location A, and begins to work as expected. 

### Expected Behavior
GetLocationAsync to return...
New location with brand new timestamp, new latitude, new longitude

### Actual Behavior
GetLocationAsync is returning
a new location - new timestamp, old latitude, old longitude.

**Comment by Bejasc** [comment]
For my own reference, this _may_ be similar to these issues..

https://github.com/xamarin/Essentials/issues/900
https://github.com/xamarin/Essentials/issues/551

**Comment by mina5500** [comment]
same here

**Comment by charmosz** [comment]
Hi. Same problem here reported by multiple users. Is there a fix ?

**Comment by mina5500** [comment]
If you increase timeout to 10 seconds... it works fine... if you set to 1 second and the device didn't get the location it gets the last known location while it should return nothing

**Comment by mesheets** [comment]
Similar to #1678 ?

**Comment by mesheets** [comment]
While the steps to reproduce were on iOS, have also had numerous issue reports with this on Android.

In looking at the [Android implementation](https://github.com/xamarin/Essentials/blob/main/Xamarin.Essentials/Geolocation/Geolocation.android.cs), there are no calls to the underlying Android [GetCurrentLocation()](https://developer.android.com/reference/android/location/LocationManager#getCurrentLocation(java.lang.String,%20android.location.LocationRequest,%20android.os.CancellationSignal,%20java.util.concurrent.Executor,%20java.util.function.Consumer%3Candroid.location.Location%3E)), only to [GetLastKnownLocation()](https://developer.android.com/reference/android/location/LocationManager#getLastKnownLocation(java.lang.String)), which—based on some platform-specific testing—is a likely cause of this issue.

In [Xamarin.Essentials.Geolocation](https://docs.microsoft.com/en-us/dotnet/api/xamarin.essentials.geolocation?view=xamarin-essentials), when calling [Xamarin.Essentials.Geolocation.GetLocationAsync()](https://docs.microsoft.com/en-us/dotnet/api/xamarin.essentials.geolocation.getlocationasync?view=xamarin-essentials#Xamarin_Essentials_Geolocation_GetLocationAsync) [which is supposed to "returns the current location of the device"], the underlying Android API call should be to Android.Location.LocationManager.GetCurrentLocation() [and _not_ Android.Location.LocationManager.GetLastKnownLocation()].

[Android.Location.LocationManager.GetLastKnownLocation()](https://developer.android.com/reference/android/location/LocationManager#getLastKnownLocation(java.lang.String)) should only be invoked when [Xamarin.Essentials.Geolocation.GetLastKnownLocationAsync()](https://docs.microsoft.com/en-us/dotnet/api/xamarin.essentials.geolocation.getlastknownlocationasync?view=xamarin-essentials#Xamarin_Essentials_Geolocation_GetLastKnownLocationAsync) is called.

**Comment by mina5500** [comment]
@jamesmontemagno can you please help ?

**Comment by AmrAlSayed0** [comment]
@mesheets This overload of [getCurrentLocation](https://developer.android.com/reference/android/location/LocationManager#getCurrentLocation(java.lang.String,%20android.os.CancellationSignal,%20java.util.concurrent.Executor,%20java.util.function.Consumer%3Candroid.location.Location%3E)) is only available on API and this [overload](https://developer.android.com/reference/android/location/LocationManager#getCurrentLocation(java.lang.String,%20android.location.LocationRequest,%20android.os.CancellationSignal,%20java.util.concurrent.Executor,%20java.util.function.Consumer%3Candroid.location.Location%3E)) is only available for API 31 so they can't be use on all versions.

Also GetLastKnownLocation is only use when there is no [appropriate provider for the accuracy you requested](https://github.com/xamarin/Essentials/blob/main/Xamarin.Essentials/Geolocation/Geolocation.android.cs#L55) otherwise [a listener for the location is set up](https://github.com/xamarin/Essentials/blob/main/Xamarin.Essentials/Geolocation/Geolocation.android.cs#L71) to get the live location

**Comment by mesheets** [comment]
Yes, I am aware that one overload of GetCurrentLocation() was introduced in API level 30 and another in API level 31; however, there still is no exposing of this method of this method even if these newer API levels are in use.

To the main problem of this issue that has been reported, though, outdated location information is returned (and not the "live" location), so something is not working.  In internal troubleshooting, making a call GetCurrentLocation() does force an update, but there is currently no way to use that via Xamarin.Essentials.
